addons/mail/models/res_partner.py

```diff
@@ -136,7 +136,18 @@ def _find_or_create_from_emails(self, emails, additional_values=None):
         :return: res.partner records in a list, following order of emails. It
           is not a recordset, to keep Falsy values.
         """
-        additional_values = additional_values if additional_values else {}
+        # Transforms additional_values keys, name_addr format into email normalized format (if given name_addr format)
+        if additional_values:
+            name_addr_pattern = r'^.*?\s*<([^<>]+)>$'
+            for key in list(additional_values.keys()):
+                if isinstance(key, str) and (match := re.match(name_addr_pattern, key)):
+                    normalized_email = match.group(1).strip().lower()
+                    if normalized_email in additional_values:
+                        raise ValueError(f"Duplicate normalized email key found: {normalized_email}")
+                    additional_values[normalized_email] = additional_values.pop(key)
```

Comment:
Doesn't email_normalize from odoo.tools work for this? Usually you don't want to be rolling your own regex for standard stuff like this